### PR TITLE
Implement entity regions

### PR DESCRIPTION
**Pull request by NotStirred**
Previously I'd wanted to wait until I'd moved CubicChunks support out, and could rewrite the region parsing altogether.

This implementation isn't ideal but should be fine for now.

Ideally the first three commits should be squashed with the last being separate, let me know if I should do that or if it'll be done in the merge

image

### CODE REVIEW

**Review by ShirleyNekoDev — COMMENTED** [review]
Some changes here might have minor conflicts with #1478, which can be fixed here after merging it, but nothing serious as far as I can see. Nice to have paintings back 🎉

**Comment by NotStirred** [comment]
I see - I'll rebase onto master once #1478 is merged.
I'll also review it to hopefully inspire a merge xd

**Comment by leMaik** [comment]
Thanks @ShirleyNekoDev and also for the code review @NotStirred. :+1:
I'll merge this once it's rebased. I'd also like to backport this PR into the next 2.4 release; could you re-open this PR against the chunky-2.4.x branch? I'm not sure if I should backport #1478.

**Comment by NotStirred** [comment]
Not sure about backporting #1478. It would be nice to have and I don't see much possibility for bugs, but it is an unnecessary change for an old version, as it's just a refactor.

Let me know if you're planning on it though, as I'll wait until that PR is opened/merged before backporting this one

**Comment by leMaik** [comment]
@NotStirred Allright, I won't backport #1478 then.

File: chunky/src/java/se/llbit/chunky/world/region/MCRegion.java
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/world/region/MCRegion.java:289]: Nice! :clap:

File: chunky/src/java/se/llbit/chunky/world/region/MCRegion.java
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/world/region/MCRegion.java:302]: This change doesn't seem right. Is it bad if the sectorOffset is 0 but the sectorCount is not zero?
```suggestion
    if (sectorOffset == 0 && sectorCount == 0) {
```

Comment (NotStirred) [reply chunky/src/java/se/llbit/chunky/world/region/MCRegion.java:302]: If either is zero, the header entry is invalid.
Was concerned that we would be running through the later code with an invalid header entry.

Comment (NotStirred) [reply chunky/src/java/se/llbit/chunky/world/region/MCRegion.java:302]: Also now that I think about it, a `sectorOffset` of `1` is also invalid as that points to the timestamp sector.
We probably don't need to think about that though.

**Review by leMaik — APPROVED** [review]

**Pull request merged**
